packages/hardware-trezor/src/transformers/assets.ts

```diff
@@ -0,0 +1,51 @@
+import * as Trezor from 'trezor-connect';
+import { Cardano } from '@cardano-sdk/core';
+
+const compareAssetNameCanonically = (a: Trezor.CardanoToken, b: Trezor.CardanoToken) => {
+  if (a.assetNameBytes === b.assetNameBytes) {
+    return a.assetNameBytes > b.assetNameBytes ? 1 : -1;
+  } else if (a.assetNameBytes.length > b.assetNameBytes.length) return 1;
+  return -1;
+};
+
+const comparePolicyIdCanonically = (a: Trezor.CardanoAssetGroup, b: Trezor.CardanoAssetGroup) => {
+  if (a.policyId === b.policyId) {
+    return a.policyId > b.policyId ? 1 : -1;
+  } else if (a.policyId.length > b.policyId.length) return 1;
+  return -1;
+};
+
+const tokenMapToAssetGroup = (tokenMap: Cardano.TokenMap): Trezor.CardanoAssetGroup[] => {
+  const map = new Map<string, Array<Trezor.CardanoToken>>();
+
+  for (const [key, value] of tokenMap.entries()) {
+    const policyId = Cardano.AssetId.getPolicyId(key);
+    const assetName = Cardano.AssetId.getAssetName(key);
+
+    if (!map.has(policyId)) map.set(policyId, new Array<Trezor.CardanoToken>());
+
+    map.get(policyId)!.push({
+      amount: value.toString(),
+      assetNameBytes: assetName
+    });
+  }
+
+  const tokenMapAssetsGroup = [];
+  for (const [key, value] of map.entries()) {
+    value.sort(compareAssetNameCanonically);
+    tokenMapAssetsGroup.push({
+      policyId: key,
+      tokenAmounts: value
+    });
+  }
+
+  tokenMapAssetsGroup.sort(comparePolicyIdCanonically);
+
+  return tokenMapAssetsGroup;
+};
+
+export const mapTokenMap = (tokenMap: Cardano.TokenMap | undefined) => {
+  if (!tokenMap) return null;
+
+  return tokenMapToAssetGroup(tokenMap);
```

Comment:
could be simplified to a one-liner function:
```
export const mapTokenMap = (tokenMap: Cardano.TokenMap | undefined) => tokenMap ? tokenMapToAssetGroup(tokenMap) : null;
```

Reply:
Done in [f3c53ef](https://github.com/input-output-hk/cardano-js-sdk/pull/793/commits/f3c53ef6517130a3c1a8525a2b6cac4961be9f01)